core-blocks/file/index.js

```diff
@@ -0,0 +1,172 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { createBlobURL } from '@wordpress/blob';
+import { createBlock } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import edit from './edit';
+
+export const name = 'core/file';
+
+export const settings = {
+	title: __( 'File' ),
+
+	description: __( 'Add a link to a file that visitors can download.' ),
+
+	icon: 'media-default',
+
+	category: 'common',
+
+	keywords: [ __( 'document' ), __( 'pdf' ) ],
+
+	attributes: {
+		id: {
+			type: 'number',
+		},
+		href: {
+			type: 'string',
+		},
+		fileName: {
+			type: 'string',
+			source: 'text',
+			selector: 'a:not([download])',
+		},
+		// Differs to the href when the block is configured to link to the attachment page
+		textLinkHref: {
+			type: 'string',
+			source: 'attribute',
+			selector: 'a:not([download])',
+			attribute: 'href',
+		},
+		// e.g. `_blank` when the block is configured to open in a new window
+		textLinkTarget: {
+			type: 'string',
+			source: 'attribute',
+			selector: 'a:not([download])',
+			attribute: 'target',
+		},
+		showDownloadButton: {
+			type: 'boolean',
+			default: true,
+		},
+		downloadButtonText: {
+			type: 'string',
+			source: 'text',
+			selector: 'a[download]',
+			default: __( 'Download' ),
+		},
+	},
+
+	supports: {
+		align: true,
+	},
+
+	transforms: {
+		from: [
+			{
+				type: 'files',
```

Comment:
Seems to me this should have been given a `priority` of low value, since as implemented it has identical priority as any other file transform, when in fact this should be the fallback. For example, dropping an image should take advantage of the Image block's `files` transform before this one. I don't know if this is currently working, and if it is it only incidental and prone to future breakage.

See: https://wordpress.org/gutenberg/handbook/block-api/#transforms-optional

>To control the priority with which a transform is applied, define a priority numeric property on your transform object, where a lower value will take precedence over higher values. This behaves much like a WordPress hook. Like hooks, the default priority is 10 when not otherwise set.

Reply:
Issue at #8022